packages/examples/src/debugger/client/debugger.ts

```diff
@@ -0,0 +1,112 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2024 TypeFox and others.
+ * Licensed under the MIT License. See LICENSE in the package root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import type { ExtensionConfig } from 'monaco-editor-wrapper';
+import type { ConfigParams, InitMessage } from '../common/definitions.js';
+
+// This is derived from:
+// https://github.com/CodinGame/monaco-vscode-api/blob/main/demo/src/features/debugger.ts
+// The client configuration is generic and can be used for a another language
+
+export const provideDebuggerExtensionConfig = (config: ConfigParams): ExtensionConfig => {
+    const filesOrContents = new Map<string, string | URL>();
+    filesOrContents.set('./extension.js', '// nothing');
+
+    return {
+        config: {
+            name: config.extensionName,
+            publisher: 'TypeFox',
+            version: '1.0.0',
+            engines: {
+                vscode: '*'
+            },
+            // A browser field is mandatory for the extension to be flagged as `web`
+            browser: 'extension.js',
+            contributes: {
+                debuggers: [
+                    {
+                        type: config.languageId,
+                        label: 'Test',
+                        languages: [config.languageId]
+                    }
+                ],
+                breakpoints: [
+                    {
+                        language: config.languageId
+                    }
+                ]
+            },
+            activationEvents: [
+                'onDebug'
+            ]
+        },
+        filesOrContents
+    };
+};
+
+export const confiugureDebugging = async (api: typeof vscode, config: ConfigParams) => {
+    class WebsocketDebugAdapter implements vscode.DebugAdapter {
+        private websocket: WebSocket;
+
+        constructor(websocket: WebSocket) {
+            this.websocket = websocket;
+            this.websocket.onmessage = (message) => {
+                this._onDidSendMessage.fire(JSON.parse(message.data));
+            };
+        }
+
+        _onDidSendMessage = new api.EventEmitter<vscode.DebugProtocolMessage>();
+        onDidSendMessage = this._onDidSendMessage.event;
+
+        handleMessage(message: vscode.DebugProtocolMessage): void {
+            const msg = JSON.stringify(message).replaceAll('\\\\', '/');
```

Comment:
@CGNonofr that is really the fix/WA. I have a path with slashes (unix style), because I define them as such when I register them. When the come out here They are `\\\\` instead of `/`. I use Chrome on Windows, but I see no reason why it should behave as such.

Reply:
I will test this with Ubuntu now

Reply:
On Ubuntu it is correct. That is not properly handled in vscode then arggghh💥 
![image](https://github.com/user-attachments/assets/38e88731-a744-441d-ac47-bc76a430601b)

Reply:
It looks like a hack though :-\  VSCode is made by microsoft so I doubt they didn't think of testing on windows ? :thinking:

Reply:
I know, but creating a breakpoint via click with Chrome on Windows and on Linux creates different source path. We have to do it like this for now.

Reply:
Does that debugger work with VSCode on Windows?

Reply:
I haven't had the time to test this, yet

Reply:
This is an example. I vote for going with the hack now and investigate properly why later and then fix later. I can add a comment here as clarification.

Reply:
Yeah I was going to suggest at least a comment :)